Woof/Woof/Model/Preferences/PreferencesHandler.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+
+/**
+ A type used for managing user preferences in an application.
+ */
+enum PreferencesHandler {
+    /**
+     Saves the selected role as a user preference.
+
+     - Parameters:
+        - selectedRole: The selected role to be saved.
+
+     - Returns: A boolean value indicating whether the saving operation was successful.
+     */
+    @discardableResult static func saveSelectedRole(_ selectedRole: Role) -> Bool {
+        var preferences: Preferences = loadPreferencesFromStorage() ?? Preferences()
+
+        preferences.selectedRole = selectedRole
+
+        return savePreferencesToStorage(preferences)
+    }
+
+    /**
+     Loads the selected role from user preferences.
+
+     - Returns: The selected role, or `.none` if no preferences are found.
+     */
+    static func loadSelectedRole() -> Role {
+        guard let preferences = loadPreferencesFromStorage() else { return .none }
+
+        let selectedRole = preferences.selectedRole
+
+        return selectedRole
+    }
+    
+    // MARK: - Private interface
+    
+    private static func loadPreferencesFromStorage() -> Preferences? {
+        guard let data = KeyValueStorage(KeyValueStorage.Name.preferences)
```

Comment:
Also shouldn't the `KeyValueStorage.loadData()` method accept the `KeyValueStorage.Key` type to avoid "accidental" storages?

Reply:
> Also shouldn't the KeyValueStorage.loadData() method accept the KeyValueStorage.Key type to avoid "accidental" storages?

Thanks for highlighting the weak point, creating an issue and discussing the refactor of this method at the next sprint planning

> to improve readability.

Thanks, i'll fix it

Reply:
fixed